hepcrawl/parsers/jats.py

```diff
@@ -375,9 +375,11 @@ def get_author_affiliations(self, author_node):
         for raw_referred_id in raw_referred_ids:
             referred_ids.update(set(raw_referred_id.split(' ')))
 
-        affiliations = [self.get_affiliation(rid) for rid in referred_ids]
+        affiliations = filter(None, (
+            self.get_affiliation(rid) for rid in referred_ids)
```

Comment:
`[self.get_affiliation(rid) for rid in referred_ids if self.get_affiliation(rid)]` is more explicit.